packages/web/src/utils/platform.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ *
+ * Copyright 2020-2025 Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+export interface PlatformBasicInfo {
+	'user_agent.language': string
+	'user_agent.mobile'?: boolean
+	'user_agent.original': string
+	'user_agent.os.name': string
+}
+export interface PlatformInfo extends PlatformBasicInfo {
+	'user_agent.os.version'?: string
+}
+
+/**
+ * Gets basic platform information synchronously
+ */
+export function getBasicPlatformInfo(): PlatformBasicInfo | undefined {
+	if (typeof navigator === 'undefined') {
+		return undefined
+	}
+
+	const userAgentData = (navigator as any).userAgentData
```

Comment:
Please avoid using any. Create a definition file and define all properties that are missing on the navigator object. Type them as optional.